### DISCUSSION TITLE
proteomics-specific tab

### DISCUSSION THREAD

**Discussion by mtiessen1175** [ndcn/omicser#190] [Ideas]
Hi to all, I'm Max, a working student in the Ori-lab at FLI. 

As discussed in #187 , there is the idea to integrate a tab in the "Playground" for lab-specific visualizations. Here we want to start including specific visualizations for Proteomics data.
As we were discussing with @ergonyc, this tab may be realized such that it is domain-specific, meaning its content depends on the data-source that was provided at start. 
I'll try to keep you updated on the progress.

**Comment by ddifraia** [comment]
@ergonyc @ricoderks  here we will give update on the work we are carrying together with Max!

**Reply by ergonyc** [reply]
@mtiessen1175 let us know how things are going!

**Reply by ddifraia** [reply]
@ergonyc  From our side we are still in the process of getting a little bit familiar with the code. We just need to play a bit and get a real feel on how to access the data.

**Comment by mtiessen1175** [comment]
@ergonyc, what would be the preferred way to push changes to the repository? Creating a new branch or pushing to the main-dev branch?

**Reply by ergonyc** [reply]
Hi Max,
Creating a pull request on the main-dev branch should work.  That will give us a link to your branch so we can see whats going on and we can test it together prior to merging.

**Comment by ddifraia** [comment]
Hi, @ricoderks @ergonyc 
as you may already know, together with @mtiessen1175 we are working toward developing the 'proteomic' specific tab.
We found out that some of the visualization that we need could be possibly very relevant also for all of you (read, other data types).

For now we are working on:
- An "improved" version of the volcano. Show differential expression analysis, and highlight specific features based on the annotation column that a used can select from the `rvdata` variable. (In our case we talk about proteins, but of course is automatically extended to all the rest). In the annotation each feature (protein, genes) can have multiple annotations, (e.g, GOannotation and so on) so the user should also select in a control panel the specific character (es ';',  ',' and so on) by which each annotation are distinguished. 

I reported this to your attention, because I think this could be very relevant for all -omic dataset that future users will ingest. And probably be extended _outside_ the proteomic specific panel.

@ergonyc Once we will have a more stable version of this new interface we will submit a pull request on the dev branch. 
Cheers,

Domenico

**Reply by ricoderks** [reply]
Hi @ddifraia, @mtiessen1175,

That sounds very interesting!!! Looking forward to it.

Cheers, Rico

**Reply by mtiessen1175** [reply]
The specific tab is implemented (until now only on my fork of the omicser-app) and tested with an example dataset. The result looks as follows:

![Proteomic-specific-tab](https://user-images.githubusercontent.com/27058247/162773953-189039c9-3e17-461e-ba1b-6ae4b5f55a74.png)

**Reply by ddifraia** [reply]
Awesome!! Great job! :D @mtiessen1175

**Reply by ricoderks** [reply]
Sweet!! Looks very nice!!!!
